src/convert.jl

```diff
@@ -213,6 +213,85 @@ end
 convert(::Type{HPolytope}, P::HPolytope) = P
 convert(::Type{VPolytope}, P::VPolytope) = P
 
+"""
+    convert(::Type{HPolytope}, Z::Zonotope{N}) where {N<:Real}
+
+Convert from zonotope to polytope in H-representation.
+
+### Input
+
+- `type` -- target type
+- `Z`    -- source zonotope
+
+### Output
+
+The zonotope represented as polytope in H-representation.
+
+### Notes
+
+The algorithm assumes that no generator is redundant.
+The result has ``2 \binom{p}{n-1}`` (with ``p`` being the number of generators
+and ``n`` being the ambient dimension) constraints, which is optimal under this
+assumption.
+
+### Algorithm
+
+We follow the algorithm presented in *Althoff, Stursberg, Buss: Computing
+Reachable Sets of Hybrid Systems Using a Combination of Zonotopes and Polytopes.
+2009.*
+
+The one-dimensional case is not covered by that algorithm; we manually handle
+this case, assuming that there is only one generator.
+"""
+function convert(::Type{HPolytope}, Z::Zonotope{N}) where {N<:Real}
+    p = ngens(Z)
+    n = dim(Z)
+    if p < n
+        error("can only convert a zonotope of order >= 1")
+    end
+
+    G = Z.generators
+    m = binomial(p, n - 1)
+    constraints = Vector{LinearConstraint{N}}(undef, 2 * m)
+
+    # special handling of 1D case
+    if n == 1
+        if p > 1
+            error("1D-zonotope conversion only supports a single generator")
+        end
+
+        c = Z.center[1]
+        g = G[:, 1][1]
+        constraints[1] = LinearConstraint([N(1)], c + g)
+        constraints[2] = LinearConstraint([N(-1)], g - c)
+        return HPolytope(constraints)
+    end
+
+    @assert N <: AbstractFloat "cannot convert rational zonotopes"
+    i = 0
+    c = Z.center
+    for columns in StrictlyIncreasingIndices(p, n-1)
+        i += 1
+        c⁺ = cross_product(view(G, :, columns))
+        c⁺ /= norm(c⁺, 2)
+
+        Δd = 0
+        for v in 1:p
+            Δd += abs(dot(c⁺, G[:, v]))
+        end
+
+        d⁺ = dot(c⁺, c) + Δd
+        c⁻ = -c⁺
+        d⁻ = dot(c⁻, c) + Δd
+
+        constraints[i] = LinearConstraint(c⁺, d⁺)
+        constraints[i + p] = LinearConstraint(c⁻, d⁻)
+    end
+    @assert i == m
```

Comment:
This assertion is for debugging purposes? What happens if it not satisfied?

Reply:
Then the code is broken :smile:
I can turn it into a comment if you want.

Reply:
:) i don't know, if you want to keep the assertion would you write a message to explain what is wrong? if the code was broken it would be easier to understand what went wrong. (i'm assuming that somewhere in the input is wrong, i haven't looked into details yet).